tests/test_items.py

```diff
@@ -0,0 +1,208 @@
+            continue
+        assert getattr(product, field) is None
+
+
+def test_product_missing_fields():
+    for required_field in list(_PRODUCT_MIN_KWARGS):
+        incomplete_kwargs = copy(_PRODUCT_MIN_KWARGS)
+        del incomplete_kwargs[required_field]
+        with pytest.raises(TypeError):
+            Product(**incomplete_kwargs)
+
+
+def test_product_asdict_all():
+    product = Product(**_PRODUCT_ALL_KWARGS)
+    serialized_product = product.asdict()
+    assert serialized_product == {
+        "additionalProperties": [dict(name="foo", value="bar")],
+        "aggregateRating": dict(
+            bestRating=5.0,
+            ratingValue=2.5,
+            reviewCount=123,
+        ),
+        "availability": "InStock",
+        "brand": dict(name="Ka-pow"),
+        "breadcrumbs": [
+            dict(name="Level 1", url="http://example.com/level1"),
+            dict(name="Level 2", url="http://example.com/level1/level2"),
+        ],
+        "canonicalUrl": "https://example.com/product22",
+        "color": "white",
+        "currency": "USD",
+        "currencyRaw": "$",
+        "description": "Full freshness all over the fridge\n5 Conversion Modes on demand\nSuper Cooling Plus™",
+        "descriptionHtml": (
+            "<article><p>Full freshness all over the fridge</p>"
+            "<p>5 Conversion Modes on demand</p>"
+            "<p>Super Cooling Plus&trade;</p></article>"
+        ),
+        "features": ["Easily store fragile products.", "Bluetooth connectivity."],
+        "gtin": [dict(type="foo", value="bar")],
+        "images": [
+            dict(url="http://example.com/image1.png"),
+        ],
+        "mainImage": dict(url="http://example.com/image1.png"),
+        "metadata": dict(
+            dateDownloaded="2022-12-31T13:01:54Z",
+            probability=1.0,
+        ),
+        "mpn": "HSC0424PP",
+        "name": "White two-door refrigerator",
+        "price": "9999.99",
+        "productId": "A123DK9823",
+        "regularPrice": "11999.99",
+        "size": "XL",
+        "sku": "A123DK9823",
+        "style": "polka dots",
+        "url": "https://example.com/?product=product22",
+        "variants": [{}],
+    }
+
+
+def test_product_asdict_min():
+    product = Product(**_PRODUCT_MIN_KWARGS)
+    serialized_product = product.asdict()
+    assert serialized_product == {
+        "url": "https://example.com/?product=product22",
+    }
+
+
+def test_product_asdict_tuple():
+    product = Product(**_PRODUCT_MIN_KWARGS, variants=({},))
+    serialized_product = product.asdict()
+    assert serialized_product == {
+        "url": "https://example.com/?product=product22",
+        "variants": ({},),
+    }
+
+
+def test_product_asdict_tuple_empty():
+    product = Product(**_PRODUCT_MIN_KWARGS, variants=tuple())
+    serialized_product = product.asdict()
+    assert serialized_product == {
+        "url": "https://example.com/?product=product22",
+    }
+
+
+def test_product_unknown_input():
```

Comment:
we should make sure that unknown_values are preserved when exporting (and maybe also accessing) data via ItemAdapter.